core/block_svg.ts

```diff
@@ -507,6 +507,23 @@ export class BlockSvg
     this.updateCollapsed();
   }
 
+  /**
+   * Traverses child blocks to see if any of them have a warning.
+   *
+   * @returns true if any child has a warning, false otherwise.
+   */
+  private childHasWarning(): boolean {
+    const children = this.getChildren(false);
+    for (const child of children) {
+      if (child.getIcon(WarningIcon.TYPE)) {
```

Comment:
Can be combined into `if (child.getIcon(WarningIcon.TYPE) || child.childHasWarning()) return true;`